In [76]:
import pandas as pd
import numpy as np
import matplotlib
import matplotlib.pyplot as plt
%matplotlib inline
import config 
import tmdbsimple as tmdb          # tmdbsimple 2.2.0
import time
tmdb.API_KEY = config.tmdb_api_key # get TMDB API key from config.py file

In [77]:
#get Dustin Hoffman's TMDB id
search_hoffman = tmdb.Search()      
response_hoffman = search_hoffman.person(query='Dustin Hoffman')
hoffman_id = search_hoffman.results[0]['id']
hoffman_id

4483

In [78]:
#Now get hoffman's credits
search_hoffman_id = tmdb.People(hoffman_id)
response_hoffman_id = search_hoffman_id.movie_credits()
response_hoffman_id


{'cast': [{'character': 'Raymond Babbitt',
   'credit_id': '52fe423cc3a36847f800e8b7',
   'poster_path': '/vllKYkA7eYi2UMM9yWQ1JegefBh.jpg',
   'id': 380,
   'video': False,
   'vote_count': 3317,
   'adult': False,
   'backdrop_path': '/eZ4Aj1O6E95Mk5O9oKKdvfnSWzf.jpg',
   'genre_ids': [18],
   'original_language': 'en',
   'original_title': 'Rain Man',
   'popularity': 17.248,
   'title': 'Rain Man',
   'vote_average': 7.7,
   'overview': "Selfish yuppie Charlie Babbitt's father left a fortune to his savant brother Raymond and a pittance to Charlie; they travel cross-country.",
   'release_date': '1988-12-11'},
  {'character': 'Stanley Motss',
   'credit_id': '52fe4258c3a36847f8016f15',
   'poster_path': '/cGatilylRyJiCoOu5T5sAB17HKR.jpg',
   'id': 586,
   'video': False,
   'vote_count': 459,
   'adult': False,
   'backdrop_path': '/bWoNwo04ZfZoEvEF8gliyB1EAyS.jpg',
   'genre_ids': [35, 18],
   'original_language': 'en',
   'original_title': 'Wag the Dog',
   'popularity': 10.386,
 

In [79]:
hoffman_n = len(response_hoffman_id['cast'][:])
hoffman_n   #how many movies

108

In [80]:
#Put all records in actor's first part of Data Frame (ie. hoffman_df)
labels = ['title','popularity','vote_average','vote_count']
idx = [film['id'] for film in response_hoffman_id['cast'][:]]
records = []
for film in (response_hoffman_id['cast'][:]):
    record = [film['title'], film['popularity'], film['vote_average'], film['vote_count']]
    records.append(record)

hoffman_df = pd.DataFrame.from_records(records, columns=labels, index=idx)
hoffman_df

title  popularity  \
380                                              Rain Man      17.248   
586                                           Wag the Dog      10.386   
693                                      Meet the Fockers      15.987   
819                                              Sleepers      13.808   
866                                     Finding Neverland      13.419   
879                                                  Hook      17.253   
891                               All the President's Men      21.243   
994                                            Straw Dogs      12.695   
1262                                Stranger Than Fiction       8.599   
1427                     Perfume: The Story of a Murderer      12.963   
1599                                    I Heart Huckabees       6.799   
10047             The Messenger: The Story of Joan of Arc      14.306   
10153                                              Sphere      11.832   
10199                              The Tale of Despereaux      10.544   
2284                       Mr. Magorium's Wonder Emporium      10.644   
10518                                        Marathon Man       9.702   
10551                                     Family Business       6.390   
3116                                      Midnight Cowboy      11.826   
3148                                   Madigan's Millions       0.701   
10699                                                Hero       8.744   
10743                                          Confidence       8.221   
11040                                      Little Big Man       8.654   
11329                                        Runaway Jury      11.720   
5924                                             Papillon      14.757   
322520                                                Hal       2.440   
31005                                      Moonlight Mile       6.242   
31200                                              Agatha       2.370   
23397                                       Straight Time       4.577   
6950                                             Outbreak      14.153   
31546                                    American Buffalo       2.828   
...                                                   ...         ...   
23544                                          The Point!       1.527   
1581                                          The Holiday      13.552   
435860                          Arthur Penn: The Director       0.600   
114524             Common Threads: Stories from the Quilt       0.600   
456698     Michael Ballhaus - Eine Reise durch mein Leben       0.600   
22509                              The Devil's Arithmetic       3.264   
2800                                      Rumor Has It...       7.978   
461569                                     The Star Wagon       0.600   
274213                     The Journey of the Fifth Horse       0.600   
410205  All Governments Lie: Truth, Deception, and the...       1.067   
469940                  Mantrap: Straw Dogs—The Final Cut       1.092   
479530                A Better Man: The Making of Tootsie       0.600   
467062                                          Spielberg       6.605   
504420                                            Tuesday       0.600   
514416                                 The Graduate at 25       0.876   
8592                                           Dick Tracy       9.246   
513814                        The Crowd Around the Cowboy       1.341   
527619  After Midnight: Reflecting on a Classic 35 Yea...       0.600   
527620  Controversy and Acclaim: The Timelessness of a...       0.600   
149395               Waldo Salt: A Screenwriter's Journey       0.600   
514638                              The Making of Tootsie       0.600   
574945                        On Location: Dustin Hoffman       0.626   
588296                Arthur Miller: A Man of His Century       0.600   
602545                           The 

In [81]:
#Now get 2nd part of actor's df from the tmdb.Movies for merging
labels = ['revenue','budget','profit','genres']
film_indices = [film[0] for film in hoffman_df.iterrows()]
#print(film_indices)
records = []
request_rate_limit = 40     #TMDB has rate limit of 40 requests per 10 seconds

for film_idx in film_indices:
    request_rate_limit -= 1
    if request_rate_limit == 0:            
        print("Hold yer horses, cowboy!")
        print("...wait per request limit...")
        time.sleep(10)
        request_rate_limit = 39
    movie = tmdb.Movies(film_idx)  #Each movie has unique identifier stored as index of df
    response = movie.info()
        
    profit = response['revenue'] - response['budget']
    record = [response['revenue'], response['budget'], profit, response['genres']]
    
    records.append(record)
#    if request_rate_limit <37: break

hoffman_movies_df = pd.DataFrame.from_records(records, columns=labels, index=film_indices)
hoffman_movies_df    


Hold yer horses, cowboy!
...wait per request limit...
Hold yer horses, cowboy!
...wait per request limit...


revenue    budget     profit  \
380     412800000  25000000  387800000   
586      64256513  15000000   49256513   
693     516642939  80000000  436642939   
819     165615285  44000000  121615285   
866     116766556  25000000   91766556   
879     300854823  70000000  230854823   
891      70600000   8500000   62100000   
994       3251794   2200000    1051794   
1262     53653224  30000000   23653224   
1427    132180323  50000000   82180323   
1599            0  22000000  -22000000   
10047    66976317  85000000  -18023683   
10153    37020277  80000000  -42979723   
10199    86900000  60000000   26900000   
2284     69474661         0   69474661   
10518    21709020   6500000   15209020   
10551    12195695         0   12195695   
3116     44785053   3600000   41185053   
3148            0         0          0   
10699           0  42000000  -42000000   
10743    23014206  15000000    8014206   
11040    31559552  15000000   16559552   
11329    80154140  60000000   20154140   
5924     53267000  12000000   41267000   
322520          0         0          0   
31005    10011050  21000000  -10988950   
31200           0         0          0   
23397     9900000         0    9900000   
6950    189859560  50000000  139859560   
31546      665450         0     665450   
...           ...       ...        ...   
23544           0         0          0   
1581    194168700  85000000  109168700   
435860          0         0          0   
114524          0         0          0   
456698          0         0          0   
22509           0         0          0   
2800     88933562  70000000   18933562   
461569          0         0          0   
274213          0         0          0   
410205          0         0          0   
469940          0         0          0   
479530          0         0          0   
467062          0         0          0   
504420          0         0          0   
514416          0         0          0   
8592    103738726  47000000   56738726   
513814          0         0          0   
527619          0         0          0   
527620          0         0          0   
149395          0         0          0   
514638          0         0          0   
574945          0         0          0   
588296          0         0          0   
602545          0         0          0   
458074          0         0          0   
617295          0         0          0   
617297          0         0          0   
29047     1434436     40000    1394436   
253639          0         0          0   
371556          0         0          0   

                                                   genres  
380                         [{'id': 18, 'name': 'Drama'}]  
586     [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...  
693     [{'id': 35, 'name': 'Comedy'}, {'id': 10749, '...  
819     [{'id': 80, 'name': 'Crime'}, {'id': 18, 'name...  
866                         [{'id': 18, 'name': 'Drama'}]  
879     [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...  
891     [{'id': 18, 'name': 'Drama'}, {'id': 36, 'name...  
994     [{'id': 80, 'name': 'Crime'}, {'id': 18, 'name...  
1262    [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...  
1427    [{'id': 80, 'name': 'Crime'}, {'id': 14, 'name...  
1599    [{'id': 35, 'name': 'Comedy'}, {'id': 10749, '...  
10047   [{'id': 12, 'name': 'Adventure'}, {'id': 18, '...  
10153   [{'id': 878, 'name': 'Science Fiction'}, {'id'...  
10199   [{'id': 12, 'name': 'Adventure'}, {'id': 16, '...  
2284    [{'id': 35, 'name': 'Comedy'}, {'id': 10751, '...  
10518   [{'id': 53, 'name': 'Thriller'}, {'id': 18, 'n...  
10551   [{'id': 28, 'name': 'Action'}, {'id': 35, 'nam...  
3116                        [{'id': 18, 'name': 'Drama'}]  
3148    [{'id': 28, 'name': 'Action'}, {'id': 35, 'nam...  
10699   [{'id': 18, 'name': 'Drama'}, {'id': 35, 'name...  
10743   [{'id': 28, 'name': 'Action'}, {'id': 35, 'nam...  
11040   [{'id': 12, 'name': 'Adventure'}, {'id': 37, '...  
11329   [{'id': 18, 'name

In [133]:
print(hoffman_df.shape)
hoffman_df.head(5)

(108, 4)


title  popularity  vote_average  vote_count
380           Rain Man      17.248           7.7        3317
586        Wag the Dog      10.386           6.9         459
693   Meet the Fockers      15.987           6.2        2795
819           Sleepers      13.808           7.5        1484
866  Finding Neverland      13.419           7.3        2352

In [134]:
print(hoffman_movies_df.shape)
hoffman_movies_df.head(5)

(108, 4)


revenue    budget     profit  \
380  412800000  25000000  387800000   
586   64256513  15000000   49256513   
693  516642939  80000000  436642939   
819  165615285  44000000  121615285   
866  116766556  25000000   91766556   

                                                genres  
380                      [{'id': 18, 'name': 'Drama'}]  
586  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...  
693  [{'id': 35, 'name': 'Comedy'}, {'id': 10749, '...  
819  [{'id': 80, 'name': 'Crime'}, {'id': 18, 'name...  
866                      [{'id': 18, 'name': 'Drama'}]

In [135]:
# This builds the full df, still need to build final
hoffman_full_df = pd.merge(hoffman_df, hoffman_movies_df, left_index=True, right_index=True)
print(hoffman_full_df.shape)
hoffman_full_df

(108, 8)


title  popularity  \
380                                              Rain Man      17.248   
586                                           Wag the Dog      10.386   
693                                      Meet the Fockers      15.987   
819                                              Sleepers      13.808   
866                                     Finding Neverland      13.419   
879                                                  Hook      17.253   
891                               All the President's Men      21.243   
994                                            Straw Dogs      12.695   
1262                                Stranger Than Fiction       8.599   
1427                     Perfume: The Story of a Murderer      12.963   
1599                                    I Heart Huckabees       6.799   
10047             The Messenger: The Story of Joan of Arc      14.306   
10153                                              Sphere      11.832   
10199                              The Tale of Despereaux      10.544   
2284                       Mr. Magorium's Wonder Emporium      10.644   
10518                                        Marathon Man       9.702   
10551                                     Family Business       6.390   
3116                                      Midnight Cowboy      11.826   
3148                                   Madigan's Millions       0.701   
10699                                                Hero       8.744   
10743                                          Confidence       8.221   
11040                                      Little Big Man       8.654   
11329                                        Runaway Jury      11.720   
5924                                             Papillon      14.757   
322520                                                Hal       2.440   
31005                                      Moonlight Mile       6.242   
31200                                              Agatha       2.370   
23397                                       Straight Time       4.577   
6950                                             Outbreak      14.153   
31546                                    American Buffalo       2.828   
...                                                   ...         ...   
23544                                          The Point!       1.527   
1581                                          The Holiday      13.552   
435860                          Arthur Penn: The Director       0.600   
114524             Common Threads: Stories from the Quilt       0.600   
456698     Michael Ballhaus - Eine Reise durch mein Leben       0.600   
22509                              The Devil's Arithmetic       3.264   
2800                                      Rumor Has It...       7.978   
461569                                     The Star Wagon       0.600   
274213                     The Journey of the Fifth Horse       0.600   
410205  All Governments Lie: Truth, Deception, and the...       1.067   
469940                  Mantrap: Straw Dogs—The Final Cut       1.092   
479530                A Better Man: The Making of Tootsie       0.600   
467062                                          Spielberg       6.605   
504420                                            Tuesday       0.600   
514416                                 The Graduate at 25       0.876   
8592                                           Dick Tracy       9.246   
513814                        The Crowd Around the Cowboy       1.341   
527619  After Midnight: Reflecting on a Classic 35 Yea...       0.600   
527620  Controversy and Acclaim: The Timelessness of a...       0.600   
149395               Waldo Salt: A Screenwriter's Journey       0.600   
514638                              The Making of Tootsie       0.600   
574945                        On Location: Dustin Hoffman       0.626   
588296                Arthur Miller: A Man of His Century       0.600   
602545                           The 

In [136]:
def to_list(cell):
    l = []
    for d in cell:     #cell is list of dicts
        l.append(d['name'])
    return(l)

hoffman_full_df['genre_names'] = hoffman_full_df['genres'].map(to_list)
hoffman_full_df

title  popularity  \
380                                              Rain Man      17.248   
586                                           Wag the Dog      10.386   
693                                      Meet the Fockers      15.987   
819                                              Sleepers      13.808   
866                                     Finding Neverland      13.419   
879                                                  Hook      17.253   
891                               All the President's Men      21.243   
994                                            Straw Dogs      12.695   
1262                                Stranger Than Fiction       8.599   
1427                     Perfume: The Story of a Murderer      12.963   
1599                                    I Heart Huckabees       6.799   
10047             The Messenger: The Story of Joan of Arc      14.306   
10153                                              Sphere      11.832   
10199                              The Tale of Despereaux      10.544   
2284                       Mr. Magorium's Wonder Emporium      10.644   
10518                                        Marathon Man       9.702   
10551                                     Family Business       6.390   
3116                                      Midnight Cowboy      11.826   
3148                                   Madigan's Millions       0.701   
10699                                                Hero       8.744   
10743                                          Confidence       8.221   
11040                                      Little Big Man       8.654   
11329                                        Runaway Jury      11.720   
5924                                             Papillon      14.757   
322520                                                Hal       2.440   
31005                                      Moonlight Mile       6.242   
31200                                              Agatha       2.370   
23397                                       Straight Time       4.577   
6950                                             Outbreak      14.153   
31546                                    American Buffalo       2.828   
...                                                   ...         ...   
23544                                          The Point!       1.527   
1581                                          The Holiday      13.552   
435860                          Arthur Penn: The Director       0.600   
114524             Common Threads: Stories from the Quilt       0.600   
456698     Michael Ballhaus - Eine Reise durch mein Leben       0.600   
22509                              The Devil's Arithmetic       3.264   
2800                                      Rumor Has It...       7.978   
461569                                     The Star Wagon       0.600   
274213                     The Journey of the Fifth Horse       0.600   
410205  All Governments Lie: Truth, Deception, and the...       1.067   
469940                  Mantrap: Straw Dogs—The Final Cut       1.092   
479530                A Better Man: The Making of Tootsie       0.600   
467062                                          Spielberg       6.605   
504420                                            Tuesday       0.600   
514416                                 The Graduate at 25       0.876   
8592                                           Dick Tracy       9.246   
513814                        The Crowd Around the Cowboy       1.341   
527619  After Midnight: Reflecting on a Classic 35 Yea...       0.600   
527620  Controversy and Acclaim: The Timelessness of a...       0.600   
149395               Waldo Salt: A Screenwriter's Journey       0.600   
514638                              The Making of Tootsie       0.600   
574945                        On Location: Dustin Hoffman       0.626   
588296                Arthur Miller: A Man of His Century       0.600   
602545                           The 

In [139]:
hoffman_full_df.drop(['genres'], axis=1, inplace=True)
hoffman_full_df

title  popularity  \
380                                              Rain Man      17.248   
586                                           Wag the Dog      10.386   
693                                      Meet the Fockers      15.987   
819                                              Sleepers      13.808   
866                                     Finding Neverland      13.419   
879                                                  Hook      17.253   
891                               All the President's Men      21.243   
994                                            Straw Dogs      12.695   
1262                                Stranger Than Fiction       8.599   
1427                     Perfume: The Story of a Murderer      12.963   
1599                                    I Heart Huckabees       6.799   
10047             The Messenger: The Story of Joan of Arc      14.306   
10153                                              Sphere      11.832   
10199                              The Tale of Despereaux      10.544   
2284                       Mr. Magorium's Wonder Emporium      10.644   
10518                                        Marathon Man       9.702   
10551                                     Family Business       6.390   
3116                                      Midnight Cowboy      11.826   
3148                                   Madigan's Millions       0.701   
10699                                                Hero       8.744   
10743                                          Confidence       8.221   
11040                                      Little Big Man       8.654   
11329                                        Runaway Jury      11.720   
5924                                             Papillon      14.757   
322520                                                Hal       2.440   
31005                                      Moonlight Mile       6.242   
31200                                              Agatha       2.370   
23397                                       Straight Time       4.577   
6950                                             Outbreak      14.153   
31546                                    American Buffalo       2.828   
...                                                   ...         ...   
23544                                          The Point!       1.527   
1581                                          The Holiday      13.552   
435860                          Arthur Penn: The Director       0.600   
114524             Common Threads: Stories from the Quilt       0.600   
456698     Michael Ballhaus - Eine Reise durch mein Leben       0.600   
22509                              The Devil's Arithmetic       3.264   
2800                                      Rumor Has It...       7.978   
461569                                     The Star Wagon       0.600   
274213                     The Journey of the Fifth Horse       0.600   
410205  All Governments Lie: Truth, Deception, and the...       1.067   
469940                  Mantrap: Straw Dogs—The Final Cut       1.092   
479530                A Better Man: The Making of Tootsie       0.600   
467062                                          Spielberg       6.605   
504420                                            Tuesday       0.600   
514416                                 The Graduate at 25       0.876   
8592                                           Dick Tracy       9.246   
513814                        The Crowd Around the Cowboy       1.341   
527619  After Midnight: Reflecting on a Classic 35 Yea...       0.600   
527620  Controversy and Acclaim: The Timelessness of a...       0.600   
149395               Waldo Salt: A Screenwriter's Journey       0.600   
514638                              The Making of Tootsie       0.600   
574945                        On Location: Dustin Hoffman       0.626   
588296                Arthur Miller: A Man of His Century       0.600   
602545                           The 

In [141]:
print(hoffman_full_df.shape)
hoffman_full_df.describe()

(108, 8)


popularity  vote_average   vote_count       revenue        budget  \
count  108.000000    108.000000   108.000000  1.080000e+02  1.080000e+02   
mean     6.241528      5.613889   530.879630  5.448879e+07  1.809667e+07   
std      5.731138      2.733847  1034.010581  1.268350e+08  3.406942e+07   
min      0.600000      0.000000     0.000000  0.000000e+00  0.000000e+00   
25%      0.894750      5.575000     1.000000  0.000000e+00  0.000000e+00   
50%      5.508000      6.450000    63.500000  0.000000e+00  0.000000e+00   
75%     10.425500      7.200000   515.000000  4.779270e+07  2.125000e+07   
max     21.495000     10.000000  6402.000000  6.656923e+08  1.500000e+08   

             profit  
count  1.080000e+02  
mean   3.639213e+07  
std    1.014932e+08  
min   -4.800000e+07  
25%    0.000000e+00  
50%    0.000000e+00  
75%    1.923871e+07  
max    5.156923e+08

Still need to delete rows with non-movies (eg. Documentary(99), TV Movie(10770) ). Deciding what constitutes a genre.
1. Eliminate rows containing Documentery or TV Movie as a genre.
2. Eliminate rows where revenue or budget are zero.
3. Exclude Music(10402) from consideration because it doesn't say what kind of genre. It could be a dramatic, romantic or acomedy, and adventure etc... I have not seen a movie classified as solely music. So, perhaps don't delete it, just think of it as a secondary classification, at least initially.
4. Cluster analysis:
    A. Drama(18) - could include Crime(80), and War(10752);
    B. Comedy(35);
    C. Romance(10749);
    C. Family(10751) and Animation(16);
    D. Western(37);
    E. Action(28)/Adventure(12);
    F. Science Fiction(878);
    G. Horror(27) and Mystery(9648) and Thriller(53).

In [142]:
hoffman_full_df.head(8)

title  popularity  vote_average  vote_count    revenue  \
380                 Rain Man      17.248           7.7        3317  412800000   
586              Wag the Dog      10.386           6.9         459   64256513   
693         Meet the Fockers      15.987           6.2        2795  516642939   
819                 Sleepers      13.808           7.5        1484  165615285   
866        Finding Neverland      13.419           7.3        2352  116766556   
879                     Hook      17.253           6.9        3083  300854823   
891  All the President's Men      21.243           7.7         807   70600000   
994               Straw Dogs      12.695           7.2         406    3251794   

       budget     profit                           genre_names  
380  25000000  387800000                               [Drama]  
586  15000000   49256513                       [Comedy, Drama]  
693  80000000  436642939                     [Comedy, Romance]  
819  44000000  121615285              [Crime, Drama, Thriller]  
866  25000000   91766556                               [Drama]  
879  70000000  230854823  [Adventure, Fantasy, Comedy, Family]  
891   8500000   62100000   [Drama, History, Mystery, Thriller]  
994   2200000    1051794              [Crime, Drama, Thriller]

In [ ]:
# If Genres is in Documentary(99), TV Movie(10770), delete the row.
# If revenue, budget or profit is zero or NaN, eliminate the row.
# If vote_count	< 20, eliminate the row.


In [73]:
exclude_genre = ['TV Movie','Documentary']
delete_rows = []
list_of_genres = (row['genres'])
for genre in list_of_genres:     
    if genre['name'] in exclude_genre:
        print(genre['name'])
        print(hoffman_full_df.loc[idx])         
        delete_rows.append(idx)
print('The # of rows to exclude is: {}'.format(len(delete_rows)))
hoffman_full_df.shape

Documentary


KeyError: 371556

In [70]:
hoffman_full_df.drop(delete_rows, inplace=True)
hoffman_full_df.shape

(107, 8)

title                                           Hal
popularity                                     2.44
vote_average                                      0
vote_count                                        1
revenue                                           0
budget                                            0
profit                                            0
genres          [{'id': 99, 'name': 'Documentary'}]
Name: 322520, dtype: object
title                                           Hal
popularity                                     2.44
vote_average                                      0
vote_count                                        1
revenue                                           0
budget                                            0
profit                                            0
genres          [{'id': 99, 'name': 'Documentary'}]
Name: 322520, dtype: object
title                                           Hal
popularity                                     2.44
vote_ave

Name: 80489, dtype: object
title                         The Magnificent Rebel
popularity                                      0.6
vote_average                                    5.8
vote_count                                        4
revenue                                           0
budget                                            0
profit                                            0
genres          [{'id': 99, 'name': 'Documentary'}]
Name: 80489, dtype: object
title                         The Magnificent Rebel
popularity                                      0.6
vote_average                                    5.8
vote_count                                        4
revenue                                           0
budget                                            0
profit                                            0
genres          [{'id': 99, 'name': 'Documentary'}]
Name: 80489, dtype: object
title                         The Magnificent Rebel
popularity                         

Name: 132507, dtype: object
title                              Against the Tide
popularity                                      0.6
vote_average                                      7
vote_count                                        1
revenue                                           0
budget                                            0
profit                                            0
genres          [{'id': 99, 'name': 'Documentary'}]
Name: 132507, dtype: object
title                              Against the Tide
popularity                                      0.6
vote_average                                      7
vote_count                                        1
revenue                                           0
budget                                            0
profit                                            0
genres          [{'id': 99, 'name': 'Documentary'}]
Name: 132507, dtype: object
title                              Against the Tide
popularity                      

Name: 189203, dtype: object
title                           All the President's Men Revisited
popularity                                                  0.934
vote_average                                                  7.5
vote_count                                                      3
revenue                                                         0
budget                                                          0
profit                                                          0
genres          [{'id': 99, 'name': 'Documentary'}, {'id': 107...
Name: 189203, dtype: object
title                           All the President's Men Revisited
popularity                                                  0.934
vote_average                                                  7.5
vote_count                                                      3
revenue                                                         0
budget                                                          0
profit              

Name: 60230, dtype: object
title                                        Trumbo
popularity                                    2.596
vote_average                                    7.1
vote_count                                       10
revenue                                           0
budget                                            0
profit                                            0
genres          [{'id': 99, 'name': 'Documentary'}]
Name: 60230, dtype: object
title                                        Trumbo
popularity                                    2.596
vote_average                                    7.1
vote_count                                       10
revenue                                           0
budget                                            0
profit                                            0
genres          [{'id': 99, 'name': 'Documentary'}]
Name: 60230, dtype: object
title                                        Trumbo
popularity                         

Name: 313502, dtype: object
title               Goldwyn: The Man and His Movies
popularity                                    0.657
vote_average                                      8
vote_count                                        1
revenue                                           0
budget                                            0
profit                                            0
genres          [{'id': 99, 'name': 'Documentary'}]
Name: 313502, dtype: object
title               Goldwyn: The Man and His Movies
popularity                                    0.657
vote_average                                      8
vote_count                                        1
revenue                                           0
budget                                            0
profit                                            0
genres          [{'id': 99, 'name': 'Documentary'}]
Name: 313502, dtype: object
title               Goldwyn: The Man and His Movies
popularity                      

Name: 400639, dtype: object
title                                              The New Cinema
popularity                                                  1.095
vote_average                                                    0
vote_count                                                      0
revenue                                                         0
budget                                                          0
profit                                                          0
genres          [{'id': 10770, 'name': 'TV Movie'}, {'id': 99,...
Name: 400639, dtype: object
title                                              The New Cinema
popularity                                                  1.095
vote_average                                                    0
vote_count                                                      0
revenue                                                         0
budget                                                          0
profit              

Name: 114524, dtype: object
title                      Common Threads: Stories from the Quilt
popularity                                                    0.6
vote_average                                                    9
vote_count                                                      4
revenue                                                         0
budget                                                          0
profit                                                          0
genres          [{'id': 36, 'name': 'History'}, {'id': 99, 'na...
Name: 114524, dtype: object
title                      Common Threads: Stories from the Quilt
popularity                                                    0.6
vote_average                                                    9
vote_count                                                      4
revenue                                                         0
budget                                                          0
profit              

Name: 410205, dtype: object
title           All Governments Lie: Truth, Deception, and the...
popularity                                                  1.067
vote_average                                                  8.8
vote_count                                                      5
revenue                                                         0
budget                                                          0
profit                                                          0
genres                        [{'id': 99, 'name': 'Documentary'}]
Name: 410205, dtype: object
title           All Governments Lie: Truth, Deception, and the...
popularity                                                  1.067
vote_average                                                  8.8
vote_count                                                      5
revenue                                                         0
budget                                                          0
profit              

Name: 467062, dtype: object
title                                                   Spielberg
popularity                                                  6.605
vote_average                                                  7.5
vote_count                                                    112
revenue                                                         0
budget                                                          0
profit                                                          0
genres          [{'id': 99, 'name': 'Documentary'}, {'id': 107...
Name: 467062, dtype: object
title                                                   Spielberg
popularity                                                  6.605
vote_average                                                  7.5
vote_count                                                    112
revenue                                                         0
budget                                                          0
profit              

Name: 514416, dtype: object
title                            The Graduate at 25
popularity                                    0.876
vote_average                                      0
vote_count                                        0
revenue                                           0
budget                                            0
profit                                            0
genres          [{'id': 99, 'name': 'Documentary'}]
Name: 514416, dtype: object
title                            The Graduate at 25
popularity                                    0.876
vote_average                                      0
vote_count                                        0
revenue                                           0
budget                                            0
profit                                            0
genres          [{'id': 99, 'name': 'Documentary'}]
Name: 514416, dtype: object
title                            The Graduate at 25
popularity                      

Name: 527620, dtype: object
title           Controversy and Acclaim: The Timelessness of a...
popularity                                                    0.6
vote_average                                                    0
vote_count                                                      0
revenue                                                         0
budget                                                          0
profit                                                          0
genres                        [{'id': 99, 'name': 'Documentary'}]
Name: 527620, dtype: object
title           Controversy and Acclaim: The Timelessness of a...
popularity                                                    0.6
vote_average                                                    0
vote_count                                                      0
revenue                                                         0
budget                                                          0
profit              

Name: 514638, dtype: object
title                         The Making of Tootsie
popularity                                      0.6
vote_average                                      0
vote_count                                        0
revenue                                           0
budget                                            0
profit                                            0
genres          [{'id': 99, 'name': 'Documentary'}]
Name: 514638, dtype: object
title                         The Making of Tootsie
popularity                                      0.6
vote_average                                      0
vote_count                                        0
revenue                                           0
budget                                            0
profit                                            0
genres          [{'id': 99, 'name': 'Documentary'}]
Name: 514638, dtype: object
title                         The Making of Tootsie
popularity                      

Name: 588296, dtype: object
title                         Arthur Miller: A Man of His Century
popularity                                                    0.6
vote_average                                                    7
vote_count                                                      1
revenue                                                         0
budget                                                          0
profit                                                          0
genres          [{'id': 99, 'name': 'Documentary'}, {'id': 107...
Name: 588296, dtype: object
title                         Arthur Miller: A Man of His Century
popularity                                                    0.6
vote_average                                                    7
vote_count                                                      1
revenue                                                         0
budget                                                          0
profit              

Name: 617297, dtype: object
title           The Magic of Hollywood... Is the Magic of People
popularity                                                   0.6
vote_average                                                   0
vote_count                                                     0
revenue                                                        0
budget                                                         0
profit                                                         0
genres                       [{'id': 99, 'name': 'Documentary'}]
Name: 617297, dtype: object
title           The Magic of Hollywood... Is the Magic of People
popularity                                                   0.6
vote_average                                                   0
vote_count                                                     0
revenue                                                        0
budget                                                         0
profit                            

Name: 371556, dtype: object
title              Warren Beatty - Mister Hollywood
popularity                                    0.977
vote_average                                    3.5
vote_count                                        2
revenue                                           0
budget                                            0
profit                                            0
genres          [{'id': 99, 'name': 'Documentary'}]
Name: 371556, dtype: object
title              Warren Beatty - Mister Hollywood
popularity                                    0.977
vote_average                                    3.5
vote_count                                        2
revenue                                           0
budget                                            0
profit                                            0
genres          [{'id': 99, 'name': 'Documentary'}]
Name: 371556, dtype: object
title              Warren Beatty - Mister Hollywood
popularity                      